roachtest: jepsen/sequential/parts-start-kill-2 failed

SHA: https://github.com/cockroachdb/cockroach/commits/c11a254ed74c59ac3ff083356d9a71b82db2c7d7

Parameters:

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=jepsen/sequential/parts-start-kill-2 PKG=roachtest TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1505083&tab=artifacts#/jepsen/sequential/parts-start-kill-2

```
The test failed on branch=master, cloud=gce:
test artifacts and logs in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/20190925-1505083/jepsen/sequential/parts-start-kill-2/run_1
	jepsen.go:256,jepsen.go:322,test_runner.go:689: exit status 1

```

The test failed with:
```
08:13:00 jepsen.go:227: grabbing artifacts from controller. Tail of controller log:
08:13:00 cluster.go:315: > /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1569387937-45-n6cpu4:6 -- tail -n 100 /mnt/data1/jepsen/cockroachdb/invoke.log
	at clojure.core$deref.invoke(core.clj:2214) ~[clojure-1.8.0.jar:na]
	at clojure.core$map$fn__4785.invoke(core.clj:2644) ~[clojure-1.8.0.jar:na]
	at clojure.lang.LazySeq.sval(LazySeq.java:40) ~[clojure-1.8.0.jar:na]
	at clojure.lang.LazySeq.seq(LazySeq.java:49) ~[clojure-1.8.0.jar:na]
	at clojure.lang.RT.seq(RT.java:521) ~[clojure-1.8.0.jar:na]
	at clojure.core$seq__4357.invokeStatic(core.clj:137) ~[clojure-1.8.0.jar:na]
	at clojure.core.protocols$seq_reduce.invokeStatic(protocols.clj:24) ~[clojure-1.8.0.jar:na]
	at clojure.core.protocols$fn__6738.invokeStatic(protocols.clj:75) ~[clojure-1.8.0.jar:na]
	at clojure.core.protocols$fn__6738.invoke(protocols.clj:75) ~[clojure-1.8.0.jar:na]
	at clojure.core.protocols$fn__6684$G__6679__6697.invoke(protocols.clj:13) ~[clojure-1.8.0.jar:na]
	at clojure.core$reduce.invokeStatic(core.clj:6545) ~[clojure-1.8.0.jar:na]
	at clojure.core$into.invokeStatic(core.clj:6610) ~[clojure-1.8.0.jar:na]
	at clojure.core$into.invoke(core.clj:6604) ~[clojure-1.8.0.jar:na]
	at jepsen.control$on_nodes.invokeStatic(control.clj:373) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$on_nodes.invoke(control.clj:357) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$on_nodes.invokeStatic(control.clj:362) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$on_nodes.invoke(control.clj:357) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.net$reify__4287.heal_BANG_(net.clj:65) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.nemesis$partitioner$reify__4390.setup_BANG_(nemesis.clj:101) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.nemesis$setup_compat_BANG_.invokeStatic(nemesis.clj:30) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.nemesis$setup_compat_BANG_.invoke(nemesis.clj:23) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.nemesis$compose$reify__4403$fn__4404.invoke(nemesis.clj:179) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.util$map_vals$fn__1319.invoke(util.clj:503) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clojure.core.reducers$map$fn__554$fn__555.invoke(reducers.clj:167) ~[knossos-0.3.3.jar:na]
	at clojure.core.protocols$iter_reduce.invokeStatic(protocols.clj:49) ~[clojure-1.8.0.jar:na]
	at clojure.core.protocols$fn__6742.invokeStatic(protocols.clj:75) ~[clojure-1.8.0.jar:na]
	at clojure.core.protocols$fn__6742.invoke(protocols.clj:75) ~[clojure-1.8.0.jar:na]
	at clojure.core.protocols$fn__6684$G__6679__6697.invoke(protocols.clj:13) ~[clojure-1.8.0.jar:na]
	at clojure.core.reducers$folder$reify__537.coll_reduce(reducers.clj:126) ~[knossos-0.3.3.jar:na]
	at clojure.core$reduce.invokeStatic(core.clj:6545) ~[clojure-1.8.0.jar:na]
	at clojure.core$into.invokeStatic(core.clj:6610) ~[clojure-1.8.0.jar:na]
	at clojure.core$into.invoke(core.clj:6604) ~[clojure-1.8.0.jar:na]
	at jepsen.util$map_kv.invokeStatic(util.clj:498) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.util$map_kv.invoke(util.clj:494) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.util$map_vals.invokeStatic(util.clj:503) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.util$map_vals.invoke(util.clj:500) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.nemesis$compose$reify__4403.setup_BANG_(nemesis.clj:179) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.nemesis$setup_compat_BANG_.invokeStatic(nemesis.clj:30) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.nemesis$setup_compat_BANG_.invoke(nemesis.clj:23) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.core.NemesisWorker.setup_worker_BANG_(core.clj:434) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.core$do_worker_BANG_.invokeStatic(core.clj:172) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.core$do_worker_BANG_.invoke(core.clj:162) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.core$run_workers_BANG_$fn__4599$fn__4600.invoke(core.clj:228) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clojure.lang.AFn.applyToHelper(AFn.java:152) ~[clojure-1.8.0.jar:na]
	at clojure.lang.AFn.applyTo(AFn.java:144) ~[clojure-1.8.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:646) ~[clojure-1.8.0.jar:na]
	at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1881) ~[clojure-1.8.0.jar:na]
	at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1881) ~[clojure-1.8.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:425) ~[clojure-1.8.0.jar:na]
	at clojure.lang.AFn.applyToHelper(AFn.java:156) ~[clojure-1.8.0.jar:na]
	at clojure.lang.RestFn.applyTo(RestFn.java:132) ~[clojure-1.8.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:650) ~[clojure-1.8.0.jar:na]
	at clojure.core$bound_fn_STAR_$fn__4671.doInvoke(core.clj:1911) ~[clojure-1.8.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:397) ~[clojure-1.8.0.jar:na]
	at clojure.lang.AFn.run(AFn.java:22) ~[clojure-1.8.0.jar:na]
	at java.lang.Thread.run(Thread.java:748) ~[na:1.8.0_222]
Caused by: com.jcraft.jsch.JSchException: java.lang.NullPointerException
	at com.jcraft.jsch.Channel.connect(Channel.java:159) ~[jsch-0.1.53.jar:na]
	at com.jcraft.jsch.Channel.connect(Channel.java:145) ~[jsch-0.1.53.jar:na]
	at clj_ssh.ssh$connect_channel.invokeStatic(ssh.clj:512) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clj_ssh.ssh$connect_channel.invoke(ssh.clj:509) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clj_ssh.ssh$ssh_exec_proc.invokeStatic(ssh.clj:667) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clj_ssh.ssh$ssh_exec_proc.invoke(ssh.clj:646) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clj_ssh.ssh$ssh_exec.invokeStatic(ssh.clj:677) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clj_ssh.ssh$ssh_exec.invoke(ssh.clj:670) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clj_ssh.ssh$ssh.invokeStatic(ssh.clj:723) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clj_ssh.ssh$ssh.invoke(ssh.clj:699) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$ssh_STAR_$fn__2034.invoke(control.clj:145) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$ssh_STAR_.invokeStatic(control.clj:145) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$ssh_STAR_.invoke(control.clj:141) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$exec_STAR_.invokeStatic(control.clj:166) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$exec_STAR_.doInvoke(control.clj:163) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clojure.lang.RestFn.applyTo(RestFn.java:137) ~[clojure-1.8.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:646) ~[clojure-1.8.0.jar:na]
	at clojure.core$apply.invoke(core.clj:641) ~[clojure-1.8.0.jar:na]
	at jepsen.control$exec.invokeStatic(control.clj:182) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$exec.doInvoke(control.clj:176) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:436) ~[clojure-1.8.0.jar:na]
	at jepsen.net$reify__4287$fn__4288.invoke(net.clj:67) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at jepsen.control$on_nodes$fn__2116.invoke(control.clj:372) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clojure.lang.AFn.applyToHelper(AFn.java:154) ~[clojure-1.8.0.jar:na]
	at clojure.lang.AFn.applyTo(AFn.java:144) ~[clojure-1.8.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:646) ~[clojure-1.8.0.jar:na]
	at clojure.core$with_bindings_STAR_.invokeStatic(core.clj:1881) ~[clojure-1.8.0.jar:na]
	at clojure.core$with_bindings_STAR_.doInvoke(core.clj:1881) ~[clojure-1.8.0.jar:na]
	at clojure.lang.RestFn.applyTo(RestFn.java:142) ~[clojure-1.8.0.jar:na]
	at clojure.core$apply.invokeStatic(core.clj:650) ~[clojure-1.8.0.jar:na]
	at clojure.core$bound_fn_STAR_$fn__4671.doInvoke(core.clj:1911) ~[clojure-1.8.0.jar:na]
	at clojure.lang.RestFn.invoke(RestFn.java:408) ~[clojure-1.8.0.jar:na]
	at jepsen.util$real_pmap$launcher__1168$fn__1169.invoke(util.clj:49) ~[jepsen-0.1.9-SNAPSHOT.jar:na]
	at clojure.core$binding_conveyor_fn$fn__4676.invoke(core.clj:1938) ~[clojure-1.8.0.jar:na]
	at clojure.lang.AFn.call(AFn.java:18) ~[clojure-1.8.0.jar:na]
	at java.util.concurrent.FutureTask.run(FutureTask.java:266) ~[na:1.8.0_222]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) ~[na:1.8.0_222]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) ~[na:1.8.0_222]
	... 1 common frames omitted
Caused by: java.lang.NullPointerException: null
	at com.jcraft.jsch.ChannelExec.start(ChannelExec.java:58) ~[jsch-0.1.53.jar:na]
	at com.jcraft.jsch.Channel.connect(Channel.java:152) ~[jsch-0.1.53.jar:na]
	... 38 common frames omitted
```

But https://github.com/cockroachdb/cockroach/pull/41043 should have prevented us from failing on the error. We see that the grep worked:
```
08:13:01 cluster.go:315: > /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-1569387937-45-n6cpu4:6 -- grep -E "(Oh jeez, I'm sorry, Jepsen broke. Here's why|Caused by)" /mnt/data1/jepsen/cockroachdb/invoke.log -A1 | grep -e BrokenBarrierException -e InterruptedException -e com.jcraft.jsch.JSchException -e "clojure.lang.ExceptionInfo: clj-ssh scp failure"
Caused by: com.jcraft.jsch.JSchException: java.lang.NullPointerException
java.util.concurrent.ExecutionException: com.jcraft.jsch.JSchException: java.lang.NullPointerException
Caused by: com.jcraft.jsch.JSchException: java.lang.NullPointerException
08:13:02 jepsen.go:244: Recognized BrokenBarrier or other known exceptions (see grep output above). Ignoring it and considering the test successful. See #30527 or #26082 for some of the ignored exceptions.
```

Then we try to tar up the artifacts and fail:
```
08:13:04 test.go:284: test failure: 	jepsen.go:256,jepsen.go:322,test_runner.go:689: exit status 1
```

--- a/pkg/cmd/roachtest/jepsen.go
+++ b/pkg/cmd/roachtest/jepsen.go
@@ -251,11 +251,9 @@ cd /mnt/data1/jepsen/cockroachdb && set -eo pipefail && \
 			// -h causes tar to follow symlinks; needed by the "latest" symlink.
 			// -f- sends the output to stdout, we read it and save it to a local file.
 			"tar -chj --ignore-failed-read -f- /mnt/data1/jepsen/cockroachdb/store/latest /mnt/data1/jepsen/cockroachdb/invoke.log /var/log/")
-		output, err := cmd.Output()
-		if err != nil {
-			t.Fatal(err)
-		}
-		if err := ioutil.WriteFile(filepath.Join(outputDir, "failure-logs.tbz"), output, 0666); err != nil {
+		if output, err := cmd.Output(); err != nil {
+			t.l.Printf("failed to retrieve jepsen artifacts and invoke.log: %s", err)
+		} else if err := ioutil.WriteFile(filepath.Join(outputDir, "failure-logs.tbz"), output, 0666); err != nil {
 			t.Fatal(err)
 		}
 		if ignoreErr {

